### PR TITLE
[PWA-2227] Use sample data with personalized content

### PR DESCRIPTION
**Pull request by jeremycharron**
## Description

Front-end part for new personalized content module that generates dynamic blocks and cart rule related to a customer segment
Add discount price label in the cart + add styles for dynamic blocks banner on homepage

Test / Demo environment : https://jeremycharron-o4saqja-4txwfa4z2sor6.eu-4.magentosite.cloud/
Back-end pull-requests : https://github.com/magento-commerce/venia-sample-data-modules/pull/19 and https://github.com/magento-commerce/venia-sample-data-modules-ee/pull/2

## Related Issue

Closes PWA-2227

### Verification Stakeholders

Project maintainers

### Specification

N/A

#### Test scenario(s) for direct fix/feature

1. As a guest, verify that "Get Connected" black banner is displayed on the homepage
2. As a guest, add an item to cart and verify that LOYAL10 coupon code cannot be applied in cart.
3. Log in as sharon@example.com sample user and verify that "Welcome back" blue banner is displayed on the homepage
4. As Sharon, verify that "Get Connected" banner is not displayed on homepage
5. As Sharon, verify that LOYAL10 coupon code can be applied in cart and gives 10% off order
6. After testing step 5, verify that discount label is "Frequent Shopper 10% discount"
7. Try to install new sample data on fresh Adobe Commerce environment
8. Verify that new dynamic blocks, customer segments and cart rule are editable in Admin panel.

#### Test scenario(s) for any existing impacted features/areas

1. Verify that setup:upgrade is working and not generating any error

#### Is Browser/Device testing needed?

Yes (Chrome, Safari, Firefox, Edge) + mobile

#### Any ad-hoc/edge case scenarios that need to be considered?

Make sure that wysiwyg/welcome-back background images are being created in /pub directory

## Screenshots / Screen Captures (if appropriate)

On PWA Front-End (you need `jcharron/PWA-2227-use-sample-data-with-personalized-content` on front-end) : 
black-blue
black-banner

## Breaking Changes (if any)

No breaking change

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/CartPage/PriceSummary/discountSummary.js [hunk 1]
```diff
@@ -40,17 +41,22 @@ const getDiscount = (discounts = []) => {
...
     const classes = useStyle({}, props.classes);
     const discount = getDiscount(props.data);
+    const { formatMessage } = useIntl();
+
+    const label = discount.label
+        ? discount.label
+        : formatMessage({
+              id: 'discountSummary.lineItemLabel',
+              defaultMessage: 'Discounts applied'
+          });
```
Comment (michaelyu0123) [inline packages/venia-ui/lib/components/CartPage/PriceSummary/discountSummary.js:51]: I don't think this was part of the scope of this ticket? Was there a reason we want to replace the default message with the source label?

File: packages/venia-ui/lib/components/HomePage/homePage.module.css [hunk 2]
```diff
@@ -419,3 +419,15 @@
...
+    flex-direction: unset !important;
+    justify-content: normal !important;
+}
+
+@media (max-width: 960px) {
+    :global(.venia-home-banner-text > div) {
+        flex-direction: column !important;
+        justify-content: center !important;
+    }
+}
```
Comment (michaelyu0123) [inline packages/venia-ui/lib/components/HomePage/homePage.module.css:433]: I'm don't this is the best approach to css. By using important, it makes it harder to modify css for this in the future.

**Review by michaelyu0123 — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/CartPage/PriceSummary/discountSummary.js (hunk 1 above)
Comment (jeremycharron) [reply packages/venia-ui/lib/components/CartPage/PriceSummary/discountSummary.js:51]: It was part of the acceptance criterias to have the source label :  https://jira.corp.magento.com/browse/PWA-2227

File: packages/venia-ui/lib/components/HomePage/homePage.module.css (hunk 2 above)
Comment (jeremycharron) [reply packages/venia-ui/lib/components/HomePage/homePage.module.css:433]: I agree. I always avoid `!important`, however the original CSS on the dynamic blocks is directly added in-line with JavaScript.

The file already had many `!important` for the same reason : https://github.com/magento/pwa-studio/blob/6c8e6f638048be059846bc4b04224845089c50e7/packages/venia-ui/lib/components/HomePage/homePage.module.css

**Review by michaelyu0123 — APPROVED** [review]
LGTM, but perhaps should look into removing the js injected inline styling so we don't have to use important!

**Review by michaelyu0123 — APPROVED** [review]
LGTM

**Comment by mikhaelbois** [comment]
Some Cypress tests are failing because some images are not loading properly.
But this is not linked to the updates made here.
QA ✅

**Pull request merged**